src/nupic/utils/SdrMetrics.hpp

```diff
@@ -200,6 +200,19 @@ class ActivationFrequency : public MetricsHelper_ {
      */
     ActivationFrequency( const std::vector<UInt> dimensions, UInt period );
 
+    /**
+     * Assign an initial activation frequency to all bits in the SDR.  If this
+     * method is used then this metric will always use the exponential moving
+     * average, and skip over the regular average which is usually applied to
+     * the first "period" many samples.
+     *
+     * This method is useful for using this metric as part of boosting
+     * algorithms which seek to push the activation frequencies to a target
+     * value. These algorithms will overreact to the default early behavior of
+     * this class during the first "period" many samples.
```

Comment:
the user will have to remember to handle it, can you make it more obvious instead? Maybe returning NaN, or 0.5f for the first period samples?

Reply:
> Maybe returning NaN, or 0.5f for the first period samples?

Then the user would have to remember to skip the first period many samples.